updatecli/updatecli.d/jdks.yml

```diff
@@ -24,31 +25,61 @@ sources:
       releasetype: {{ $releasetype }}
     transformers:
       - trimprefix: "jdk{{ if (ne $major 8) }}-{{ end }}"
+{{ range $platform := splitList " " $platforms }}
+{{ $parts := splitList "/" $platform }}
+{{ $os := index $parts 0 }}
+{{ $arch := index $parts 1 }}
+  {{ $os }}_{{ $arch }}_checksum_url:
+    kind: temurin
+    spec:
+      architecture: {{ index $.mappedArch $arch }}
+      featureversion: {{ $major }}
+      operatingsystem: {{ index $.mappedOs $os }}
+      releasetype: {{ $releasetype }}
+      result: checksum_url
+  {{ $os }}_{{ $arch }}_checksum_value:
+    dependson:
+      - {{ $os }}_{{ $arch }}_checksum_url
+    kind: file
+    spec:
+      {{ $filesource := print $os "_" $arch "_checksum_url" }}
+      file: '{{ source $filesource }}'
+    transformers:
+      - findsubmatch:
+          pattern: "^([a-f0-9]+)"
+  {{ $os }}_{{ $arch }}_installer_url:
+    kind: temurin
+    spec:
+      architecture: {{ index $.mappedArch $arch }}
+      featureversion: {{ $major }}
+      operatingsystem: {{ index $.mappedOs $os }}
+      releasetype: {{ $releasetype }}
+      result: installer_url
+{{ end }}
 
 conditions:
-  checkIfReleaseIsAvailableForall:
+{{ range $platform := splitList " " $platforms }}
+{{ $parts := splitList "/" $platform }}
+{{ $os := index $parts 0 }}
+{{ $arch := index $parts 1 }}
+
+  syn_checkIfReleaseIsAvailableFor{{ $platform }}:
     kind: temurin
     # checked version comes from the source
     sourceid: lastReleaseVersion
     spec:
       # release type is mandatory to support ea version (bug in updatecli)
       releasetype: {{ $releasetype }}
       platforms:
-        - linux/x64
-        - linux/aarch64
-        - windows/x64
+        - {{ index $.mappedOs $os }}/{{ index $.mappedArch $arch }}
+{{ end }}
```

Comment:
Condition are not needed since you're now iterating on the list of platform. Each platform is 1 OS and 1 arch, so the result of source if available. Need to be removed.